client/core/trade.go

```diff
@@ -810,19 +812,27 @@ func (t *trackedTrade) processAudit(msgID uint64, audit *msgjson.Audit) error {
 	auth.AuditStamp = audit.Time
 	auth.AuditSig = audit.Sig
 	proof.CounterScript = audit.Contract
+	matchTime := encode.UnixTimeMilli(int64(auth.MatchStamp))
+	reqLockTime := matchTime.Add(dex.LockTimeMaker) // counterparty = maker, their locktime = 48 hours.
 	if dbMatch.Side == order.Maker {
 		// Check that the secret hash is correct.
 		if !bytes.Equal(proof.SecretHash, auditInfo.SecretHash()) {
 			return errs.add("secret hash mismatch. expected %x, got %x", proof.SecretHash, auditInfo.SecretHash())
 		}
 		match.setStatus(order.TakerSwapCast)
 		proof.TakerSwap = []byte(audit.CoinID)
+		// counterparty = taker, their locktime = 24 hours.
+		reqLockTime = matchTime.Add(dex.LockTimeTaker)
 	} else {
 		proof.SecretHash = auditInfo.SecretHash()
 		match.setStatus(order.MakerSwapCast)
 		proof.MakerSwap = []byte(audit.CoinID)
 	}
 
+	if auditInfo.Expiration().Before(reqLockTime) {
+		return errs.add("lock time too early. Need %s, got %s", reqLockTime, auditInfo.Expiration())
+	}
```

Comment:
Which makes me realize this audit code is not following the same truncate (`matchTime.Add(dex.LockTimeTaker).UTC().Unix()`) operations and the comparison could fail because of that.